daq/stream_monitor.py

```diff
@@ -93,16 +93,18 @@ def trigger_callback(self, fd):
                 os.read(fd, 1024)
         except Exception as e:
             if fd in self.callbacks:
-                self.callbacks[fd][5].close()
-                self.forget(fd)
-            self.error_handler(fd, e, name, on_error)
+                if self.trigger_hangup(fd, e):
```

Comment:
if the handling of hangup and error are combined, in cases where docker callback throws an error, the docker test will ignore that error and the test will still pass. I'm not sure if there's an integration test for that.